osu.Game/Screens/OsuScreen.cs

```diff
@@ -120,12 +133,27 @@ public override void OnResuming(IScreen last)
             sampleExit?.Play();
             applyArrivingDefaults(true);
 
+            if (api != null)
+                api.LocalUser.Value.Status.ValueChanged += userStatusChanged;
+
+            setUserStatus(ScreenStatus);
+
             base.OnResuming(last);
         }
 
+        private void userStatusChanged(ValueChangedEvent<UserStatus> obj)
+        {
+            if (obj.NewValue?.GetType() != ScreenStatus?.GetType()) //restore the status back to this screen's status when the user status is changed back to online after having being set to DND / offline
```

Comment:
I think this would read better if you ensure all screen statuses are always of  `UserStatusOnline` type, or maybe changing what the screens can change to be a separate class *contained* by the `UserStatus`, rather than the actual status itself.

Put another way: why are the screens able to (potentially) change online state in the first place?